src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -252,6 +252,24 @@ void GcsNodeManager::HandleReportHeartbeat(const rpc::ReportHeartbeatRequest &re
   auto heartbeat_data = std::make_shared<rpc::HeartbeatTableData>();
   heartbeat_data->CopyFrom(request.heartbeat());
 
+  if (RayConfig::instance().light_heartbeat_enabled()) {
+    auto iter = node_heartbeats_.find(node_id);
```

Comment:
The logic inside if suggests abstracting a function.

Reply:
that makes sense